### PR TITLE
docs: Add missing middleware option for guestmode

### PR DESCRIPTION
**Pull request by anischihi**
### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description
I tried the guest middleware, it did not work (n redirect, no unauthenticatedOnly behaviour). Added the missing snippet, everything works like a charm. I think has been missed in the docs. You can try for yourself.

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have added tests (if possible).
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by MuhammadM1998** [comment]
I suggested this in an issue before but Zoey said this is expected when turning off the global middleware (You don't need to add this line if global middleware is enabled). Maybe a hint that this line is needed _only_ when disabling global middleware would be best?

**Comment by zoey-kaiser** [comment]
> I suggested this in an issue before but Zoey said this is expected when turning off the global middleware (You don't need to add this line if global middleware is enabled). Maybe a hint that this line is needed only when disabling global middleware would be best?

Works for me! Adding the value will not hurt in any case, but as you pointed out, could help reduce confusion. Thanks for making the adjustment ❤️

File: docs/content/3.application-side/5.guest-mode.md [hunk 1]
```diff
@@ -15,6 +15,7 @@ This behavior is useful for login pages that you don't want to be visitable by l
 Briefly summarized, you can enable guest mode on a page by passing the following configuration:
 ```ts
 definePageMeta({
+  middleware: 'auth',
```
Comment (zoey-kaiser) [inline docs/content/3.application-side/5.guest-mode.md:18]: Maybe we add a comment behind it indicating that this is not required if the `globalAppMiddleware` is enabled?

```suggestion
  middleware: 'auth', // Only required if globalAppMiddleware is disabled
```

**Review by zoey-kaiser — CHANGES_REQUESTED** [review]

File: docs/content/3.application-side/5.guest-mode.md (hunk 1 above)
Comment (anischihi) [reply docs/content/3.application-side/5.guest-mode.md]: Thanks for the suggestion, took it in the commit 🙏

**Review by zoey-kaiser — APPROVED** [review]

**Pull request merged**